src/jsp/jsp_cl.c

```diff
@@ -2174,22 +2174,42 @@ jsp_send_destroy_request_all ()
 extern int
 jsp_send_destroy_request (const SOCKET sockfd)
 {
-  int nbytes;
-  int req_code = 0x10;
-  int res = NO_ERROR;
+  OR_ALIGNED_BUF (OR_INT_SIZE) a_request;
+  char *request = OR_ALIGNED_BUF_START (a_request);
 
-  nbytes = jsp_writen (sockfd, (void *) &req_code, (int) sizeof (int));
+  or_pack_int (request, (int) SP_CODE_DESTROY);
+  int nbytes = jsp_writen (sockfd, request, (int) sizeof (int));
   if (nbytes != (int) sizeof (int))
     {
       er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_SP_NETWORK_ERROR, 1, "destroy");
       return er_errid ();
     }
 
-  tran_begin_libcas_function ();
-  res = libcas_main (sockfd);	/* jdbc call to destroy resources */
-  tran_end_libcas_function ();
+  /* read request code */
+  int code;
+  nbytes = jsp_readn (sockfd, (char *) &code, (int) sizeof (int));
+  if (nbytes != (int) sizeof (int))
+    {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_SP_NETWORK_ERROR, 1, nbytes);
+      return ER_SP_NETWORK_ERROR;
+    }
+  code = ntohl (code);
+
+  if (code == SP_CODE_DESTROY)
+    {
+      bool mode = ssl_client;
+      ssl_client = false;
+      tran_begin_libcas_function ();
+      libcas_main (sockfd);	/* jdbc call */
+      tran_end_libcas_function ();
+      ssl_client = mode;
+    }
+  else
+    {
+      /* end */ 
```

Comment:
no necessary for else ?

Reply:
If ServerSideConnection is not created in a stored procedure, entering into the main of CAS protocol, for closing cached UStatements, is not required. So ExecuteThread will send REQ_CODE_END to the CAS. And CAS will get out of the destroying state.